File: Data Structures & Algorithms/trapping-rain-water/submission-3.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        int total =0;
        for(int i=0; i<height.size(); i++){
            int l= i;
            int leftMax =i;
            int r= i;
            int rightMax = i;
            while(l>0){
                l--;
                if(height[l]>height[leftMax]){
                    leftMax = l;
                }
            }
            while(r < height.size()-1){
                r++;
                if(height[r]>height[rightMax]){
                    rightMax = r;
                }
            }
            total += max(min(height[leftMax], height[rightMax]) - height[i], 0);

        }
        return total;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l,r = 0, len(heights)-1
        m = 0
        while l<r:
            amt = min(heights[l],heights[r])*(r-l)
            m = max(m, amt)
            if heights[l]<=heights[r]:
                l+=1
            else:
                r-=1
        return m
